libsignal-service/src/pre_keys.rs

```diff
@@ -64,4 +64,5 @@ pub struct PreKeyState {
     pub signed_pre_key: SignedPreKey,
     #[serde(with = "serde_public_key")]
     pub identity_key: PublicKey,
+    pub last_resort_key: PreKeyEntity,
```

Comment:
If we merge this as-is, Whisperfish might break... Although I do intend to implement GV2 asap now, I'd like some compatibility flag, if possible.

```suggestion
    #[serde(skip_serializing_if = "Option::is_none")] 
    pub last_resort_key: Option<PreKeyEntity>,
```

and a flag `last_resort_key: bool` in the method, maybe? This hack also does not seem to be present in `Signal-Android`, (cfr. `PushServiceSocket::registerPreKeys`, `PreKeyState`) would you know what that is about?

Reply:
So, this is only for secondary devices, to not "block" the account to join groups v2 (at least, from my observations). If you look at `signal-cli` it's also done there: https://gitlab.stusta.de/AsamK/textsecure-cli/blob/067765a62f1877fced7a82fd86a800673e47df41/src/main/java/org/asamk/signal/Manager.java#L368

Reply:
If it's certainly only for secondary devices, I'd think this justifies the `Option`al and `serde` annotation, or am I getting that wrong?

Reply:
You're absolutely right, I pushed a new commit, I think this should be good to merge now!